pynecone/state.py

```diff
@@ -61,6 +61,39 @@ def __init__(self, *args, **kwargs):
         for substate in self.get_substates():
             self.substates[substate.get_name()] = substate().set(parent_state=self)
 
+        self._init_mutable_fields()
+
+    def _init_mutable_fields(self):
+        """Initialize mutable fields.
+
+        So that mutation to them can be detected by the app:
+        * list
+        """
+        for field in self.base_vars.values():
```

Comment:
This will work for the common use case - one thing we need to think deeper about is how to treat nested types.

For example, a dict with a list value, a subclass of `pc.Base` with a list field, etc. We may later need to add some sort of recursive inspection and convert everything to `PcList`. Or state clearly that we only support this use case and the mutable notice still applies for the other case.

Reply:
Created `_convert_mutable_datatypes` in e2dbc362265d0f434180bbb81fc2fd6439965e6e to recursively convert `list` to `PcList`. But now only `list` & `dict` would be iterated to recursively do the conversion.